<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Math_crypto</class>
 <widget class="QMainWindow" name="Math_crypto">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1089</width>
    <height>708</height>
   </rect>
  </property>
  <property name="maximumSize">
   <size>
    <width>16777215</width>
    <height>16777215</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Math Cryptography</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout_3">
    <item row="0" column="0">
     <layout class="QHBoxLayout" name="horizontalLayout_4">
      <item>
       <widget class="QLabel" name="cipher_lbl">
        <property name="text">
         <string>Cipher Method:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QComboBox" name="cipher_cbox">
        <item>
         <property name="text">
          <string>Shift Cipher</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Tritemius Cipher</string>
         </property>
        </item>
       </widget>
      </item>
     </layout>
    </item>
    <item row="1" column="0">
     <layout class="QHBoxLayout" name="horizontalLayout_3">
      <item>
       <widget class="QLabel" name="key_lbl">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="text">
         <string>Cipher Key:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="key_ln_edit"/>
      </item>
     </layout>
    </item>
    <item row="2" column="0">
     <widget class="QTabWidget" name="encryption_tabs">
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="encrypt_tab">
       <attribute name="title">
        <string>Encryption</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout">
        <item row="0" column="0">
         <layout class="QHBoxLayout" name="horizontalLayout_2">
          <item>
           <widget class="QPushButton" name="encrypt_btn">
            <property name="text">
             <string>Encrypt</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="decrypt_btn">
            <property name="text">
             <string>Decrypt</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="load_key_btn">
            <property name="text">
             <string>Load Key From File</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QCheckBox" name="bytes_cbox">
            <property name="text">
             <string>Raw Bytes</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="label">
            <property name="text">
             <string>Language:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QComboBox" name="lang_cbox">
            <item>
             <property name="text">
              <string>EN</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>UKR</string>
             </property>
            </item>
           </widget>
          </item>
         </layout>
        </item>
        <item row="1" column="0">
         <layout class="QHBoxLayout" name="horizontalLayout">
          <item>
           <layout class="QVBoxLayout" name="verticalLayout_5">
            <item>
             <widget class="QRadioButton" name="initial_rbtn">
              <property name="text">
               <string>Initial</string>
              </property>
              <property name="checked">
               <bool>true</bool>
              </property>
              <attribute name="buttonGroup">
               <string notr="true">savefile_btn_group</string>
              </attribute>
             </widget>
            </item>
            <item>
             <widget class="QPlainTextEdit" name="initial_txt_edit"/>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QVBoxLayout" name="verticalLayout_6">
            <item>
             <widget class="QRadioButton" name="encrypted_rbtn">
              <property name="text">
               <string>Encrypted</string>
              </property>
              <attribute name="buttonGroup">
               <string notr="true">savefile_btn_group</string>
              </attribute>
             </widget>
            </item>
            <item>
             <widget class="QPlainTextEdit" name="encrypted_txt_edit"/>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QVBoxLayout" name="verticalLayout_7">
            <item>
             <widget class="QRadioButton" name="decrypted_rbtn">
              <property name="text">
               <string>Decrypted</string>
              </property>
              <attribute name="buttonGroup">
               <string notr="true">savefile_btn_group</string>
              </attribute>
             </widget>
            </item>
            <item>
             <widget class="QPlainTextEdit" name="decrypted_txt_edit"/>
            </item>
           </layout>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="freq_table_tab">
       <attribute name="title">
        <string>Frequency Table</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_2">
        <item row="1" column="0">
         <widget class="QTableWidget" name="freq_table_widget">
          <column>
           <property name="text">
            <string>Character</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Frequency</string>
           </property>
          </column>
         </widget>
        </item>
        <item row="0" column="0">
         <widget class="QPushButton" name="print_freq">
          <property name="text">
           <string>Count Frequencies</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="cipher_spec_ops_tab">
       <attribute name="title">
        <string>Cipher Specific Operations</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_6">
        <item row="1" column="0">
         <widget class="QStackedWidget" name="cipher_specific_ops_stacked_widget">
          <property name="currentIndex">
           <number>0</number>
          </property>
          <widget class="QWidget" name="ceasar_page">
           <layout class="QGridLayout" name="gridLayout_4">
            <item row="0" column="0">
             <widget class="QPushButton" name="brute_force_btn">
              <property name="text">
               <string>Brute Force</string>
              </property>
             </widget>
            </item>
            <item row="1" column="0">
             <widget class="QTableWidget" name="brute_force_table"/>
            </item>
           </layout>
          </widget>
          <widget class="QWidget" name="tritemius_page">
           <layout class="QGridLayout" name="gridLayout_5">
            <item row="0" column="0">
             <widget class="QComboBox" name="trit_attack_combobox">
              <item>
               <property name="text">
                <string>Message Pair</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>Frequency Tables</string>
               </property>
              </item>
             </widget>
            </item>
            <item row="1" column="0">
             <layout class="QHBoxLayout" name="horizontalLayout_8">
              <item>
               <widget class="QPushButton" name="generate_rnd_key_btn">
                <property name="text">
                 <string>Generate Random Key</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="attack_trit_cipher_btn">
                <property name="text">
                 <string>Attack Trithemius Cipher</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLabel" name="key_vec_size_lbl">
                <property name="text">
                 <string>Key Vec Size:</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QSpinBox" name="key_vec_size_spinbox"/>
              </item>
             </layout>
            </item>
            <item row="2" column="0">
             <widget class="QStackedWidget" name="trithemius_attacks_stacked_widget">
              <property name="currentIndex">
               <number>0</number>
              </property>
              <widget class="QWidget" name="page">
               <layout class="QGridLayout" name="gridLayout_7">
                <item row="2" column="0">
                 <layout class="QHBoxLayout" name="horizontalLayout_5">
                  <item>
                   <widget class="QLabel" name="trir_broken_key_lbl">
                    <property name="text">
                     <string>Broken Key:</string>
                    </property>
                   </widget>
                  </item>
                  <item>
                   <widget class="QLineEdit" name="trit_broken_key_ledit"/>
                  </item>
                 </layout>
                </item>
                <item row="1" column="0">
                 <layout class="QHBoxLayout" name="horizontalLayout_6">
                  <item>
                   <layout class="QVBoxLayout" name="verticalLayout_2">
                    <item>
                     <widget class="QLabel" name="init_msg_trit_attack_lbl">
                      <property name="text">
                       <string>Decrypted Message</string>
                      </property>
                     </widget>
                    </item>
                    <item>
                     <widget class="QPlainTextEdit" name="dec_msg_trit_attack_tedit"/>
                    </item>
                   </layout>
                  </item>
                  <item>
                   <layout class="QVBoxLayout" name="verticalLayout">
                    <item>
                     <widget class="QLabel" name="enc_msg_trit_attack_lbl">
                      <property name="text">
                       <string>Encrypted Message</string>
                      </property>
                     </widget>
                    </item>
                    <item>
                     <widget class="QPlainTextEdit" name="enc_msg_tit_attack_tedit"/>
                    </item>
                   </layout>
                  </item>
                 </layout>
                </item>
               </layout>
              </widget>
              <widget class="QWidget" name="page_2">
               <layout class="QGridLayout" name="gridLayout_8">
                <item row="0" column="0">
                 <widget class="QPushButton" name="load_lang_freq_table">
                  <property name="text">
                   <string>Load Language Freq Table</string>
                  </property>
                 </widget>
                </item>
                <item row="1" column="0">
                 <layout class="QHBoxLayout" name="horizontalLayout_7">
                  <item>
                   <widget class="QTableWidget" name="lang_freq_table"/>
                  </item>
                  <item>
                   <widget class="QTableWidget" name="freq_attack_attempts_table"/>
                  </item>
                 </layout>
                </item>
               </layout>
              </widget>
             </widget>
            </item>
           </layout>
          </widget>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1089</width>
     <height>24</height>
    </rect>
   </property>
   <widget class="QMenu" name="file_menu">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="open_action"/>
    <addaction name="save_action"/>
    <addaction name="saveas_action"/>
    <addaction name="create_new_action"/>
    <addaction name="print_action"/>
    <addaction name="exit_action"/>
   </widget>
   <widget class="QMenu" name="help_menu">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="about_action"/>
   </widget>
   <addaction name="file_menu"/>
   <addaction name="help_menu"/>
  </widget>
  <action name="open_action">
   <property name="text">
    <string>Open</string>
   </property>
  </action>
  <action name="save_action">
   <property name="text">
    <string>Save</string>
   </property>
  </action>
  <action name="saveas_action">
   <property name="text">
    <string>Save As</string>
   </property>
  </action>
  <action name="create_new_action">
   <property name="text">
    <string>Create New</string>
   </property>
  </action>
  <action name="print_action">
   <property name="text">
    <string>Print</string>
   </property>
  </action>
  <action name="about_action">
   <property name="text">
    <string>About</string>
   </property>
  </action>
  <action name="english_radiob">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>EN</string>
   </property>
  </action>
  <action name="ukrainian_radiob">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>UK</string>
   </property>
  </action>
  <action name="exit_action">
   <property name="text">
    <string>Exit</string>
   </property>
  </action>
  <action name="actionBinary_Mode">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Binary Mode</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
 <buttongroups>
  <buttongroup name="savefile_btn_group">
   <property name="exclusive">
    <bool>true</bool>
   </property>
  </buttongroup>
 </buttongroups>
</ui>
